docs/resources/permission_assignment.md

```diff
@@ -0,0 +1,60 @@
+---
+subcategory: "Security"
+---
+# databricks_permission_assignment Resource
+
+These resources are invoked in the workspace context.
+
+## Example Usage
+
+In workspace context, adding account-level user to a workspace:
+
+```hcl
+resource "databricks_user" "me" {
+  user_name = "me@example.com"
+}
+
+resource "databricks_permission_assignment" "add_user" {
+  principal_id = databricks_user.me.id
+  permissions  = ["USER"]
+}
+```
+
+In workspace context, adding account-level service principal to a workspace:
+
+```hcl
+resource "databricks_service_principal" "sp" {
+  display_name = "Automation-only SP"
+}
+
+resource "databricks_permission_assignment" "add_admin_spn" {
+  principal_id = databricks_service_principal.sp.id
+  permissions  = ["ADMIN"]
+}
+```
+
+## Argument Reference
+
+The following arguments are required:
+
+* `principal_id` - Databricks ID of the user, service principal, or group. The principal ID can be retrieved using the SCIM API, or using [databricks_user](../data-sources/user.md), [databricks_service_principal](../data-sources/service_principal.md) or [databricks_group](../data-sources/group.md) data sources.
```

Comment:
If the principal is not yet assigned to the workspace, you can't use these data sources to retrieve the IDs using the workspace API, you will need to be an account admin to do that (using account API)